graphqlapi/src/main/java/io/stargate/graphql/schema/NameConversions.java

```diff
@@ -0,0 +1,251 @@
+      } else if (!isHex(c)) {
+        return -1;
+      }
+      j += 1;
+    }
+    return -1;
+  }
+
+  private static void appendHexEscape(StringBuilder target, int codePoint) {
+    target.append('x').append(Integer.toHexString(codePoint)).append('_');
+  }
+
+  private static boolean isValidGraphql(int c) {
+    return isDigit(c) || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || c == '_';
+  }
+
+  private static boolean isHex(int c) {
+    return isDigit(c) || (c >= 'a' && c <= 'f');
+  }
+
+  private static boolean isDigit(int c) {
+    return c >= '0' && c <= '9';
+  }
+
+  private static String escapeReservedPrefix(String result) {
+    for (Map.Entry<String, String> entry : RESERVED_PREFIXES.entrySet()) {
+      String prefix = entry.getKey();
+      String replacement = entry.getValue();
+      if (result.startsWith(prefix)) {
+        return replacement + result.substring(prefix.length());
+      }
+    }
+    return result;
+  }
+
+  private static String escapeReservedSuffix(String result) {
+    for (Map.Entry<String, String> entry : RESERVED_SUFFIXES.entrySet()) {
+      String suffix = entry.getKey();
+      String replacement = entry.getValue();
+      if (result.endsWith(suffix)) {
+        return result.substring(0, result.length() - suffix.length()) + replacement;
+      }
+    }
+    return result;
+  }
+
+  private static final ImmutableMap<String, String> RESERVED_GRAPHQL_NAMES =
+      ImmutableMap.<String, String>builder()
+          // We use these names for our own types:
+          .put("Mutation", "Mutatiox6e_")
+          .put("MutationConsistency", "MutationConsistencx79_")
+          .put("MutationOptions", "MutationOptionx73_")
+          .put("Query", "Querx79_")
+          .put("QueryConsistency", "QueryConsistencx79_")
+          .put("QueryOptions", "QueryOptionx73_")
+          .put("SerialConsistency", "SerialConsistencx79_")
+          // Built-in scalars can clash with a table via "FilterInput" types
+          .put("Boolean", "Booleax6e_")
+          .put("Float", "Floax74_")
+          .put("Int", "Inx74_")
+          .put("String", "Strinx67_")
+          // Custom scalars can clash with a table
+          .put("Uuid", "Uuix64_")
+          .put("TimeUuid", "TimeUuix64_")
+          .put("Inet", "Inex74_")
+          .put("Date", "Datx65_")
+          .put("Duration", "Duratiox6e_")
+          .put("BigInt", "BigInx74_")
+          .put("Counter", "Countex72_")
+          .put("Ascii", "Ascix69_")
+          .put("Decimal", "Decimax6c_")
+          .put("Varint", "Varinx74_")
+          .put("Float32", "Float32")
+          .put("Blob", "Blox62_")
+          .put("SmallInt", "SmallInx74_")
+          .put("TinyInt", "TinyInx74_")
+          .put("Timestamp", "Timestamx70_")
+          .put("Time", "Timx65_")
+          .build();
+
+  private static ImmutableMap<String, String> RESERVED_PREFIXES =
+      ImmutableMap.<String, String>builder()
+          .put("Entry", "Entrx79_")
```

Comment:
`"insert", "update", "delete"` should also be reserved.

Reply:
I think we are good. We only use those prefixes is in the `Mutation` type and they are unconditionally prepended. If we have two tables named `foo` and `insertfoo`, the corresponding fields will be `insertfoo` and `insertinsertfoo`.

Reply:
Doesn't `insertfoo()` query (table `insertfoo`) collide with `insertfoo()` mutation (table `foo`)?

Reply:
No, queries and mutations belong to different types:
```
type Mutation {
  insertfoo
  insertinsertfoo
}
type Query {
  foo
  insertfoo
}
```

Reply:
oh, nice!